stream-controller: fix maxBufLen calculation

markx opened the pull request:
### This PR will...
Fix the calculation for `maxBufLen`
### Why is this Pull Request needed?
The code doesn't seem to match the comment. If `don't buffer more than 60 MB and more than 30s` is the wanted behavior, then here it should be `min`.


### Are there any points in the code the reviewer needs to double check?
The expected behavior should be double checked.
### Resolves issues:

### Checklist

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] API or design changes are documented in API.md

markx commented:
Well, the code doesn't seem to be consistent with [this](https://github.com/video-dev/hls.js/blob/master/docs/API.md#maxbufferlength) either. Maybe the code is correct but the docs are not accurate.

johnBartos commented:
@markx I agree, the docs are confusing about the buffer parameters. Here's the relevant code:

```javascript
    if (levelBitrate) {
      maxBufLen = Math.max(8 * config.maxBufferSize / levelBitrate, config.maxBufferLength);
    } else {
      maxBufLen = config.maxBufferLength;
    }

    maxBufLen = Math.min(maxBufLen, config.maxMaxBufferLength);
```

`maxBufferSize` (default 60Mb), in bits, is use when Hls.js knows the bitrate of a level. That calculation gives us the number of seconds of that can be buffered. Otherwise, Hls.js uses the `maxBufferLength` config, which is in seconds (default 30s). `maxMaxBufferLength` is the ceiling, and is in seconds (default 600s). 


> If maxBufferLength specifies the max buffer length regardless of maxBufferSize, then why is maxBufLen calculated based on both?

It's either or, I believe you're confusing `maxBufferLength` with `maxMaxBufferLength`.

Any modifications to the docs you find helpful are welcome

markx commented:
@johnBartos  Yeah, in the beginning I found the comment and the code don't match, so I made this PR to fix the code. But after more study, I realized the the code was probably what the author really wanted. So I reverted the code change, and just deleted the misleading comment.

The docs are in accordance with the code. It's just that one-line comment.

itsjamie commented:
Thank you @markx!

Merging this, no reason to keep it hanging out.
